modules/codegen/src/smithy4s/codegen/internals/SmithyToIR.scala

```diff
@@ -569,11 +569,22 @@ private[codegen] class SmithyToIR(model: Model, namespace: String) {
           .getAllTraits()
           .asScala
           .flatMap { case (_, trt) =>
-            model
+            val refinement = model
               .getShape(trt.toShapeId)
               .asScala
               .flatMap(_.getTrait(classOf[RefinementTrait]).asScala)
               .map(rt => trt -> ExternalTypeInfo.RefinementInfo(rt))
+
+            refinement.orElse {
+              // Handle IdRefs as a refinement
+              if (trt.toShapeId == IdRefTrait.ID) {
+                val rt = RefinementTrait
+                  .builder()
+                  .targetType("smithy4s.ShapeId")
+                  .build()
+                Some(trt -> ExternalTypeInfo.RefinementInfo(rt))
+              } else None
```

Comment:
suggestion: extract this to a parameterless `def` (for laziness) so that the `orElse` call is a simple one-liner